### PR TITLE
[FIX] stock_account: _action_done performance

### PR DESCRIPTION
**Pull request by sle-odoo**

### CODE REVIEW

File: addons/stock_account/models/product.py [hunk 1]
```diff
@@ -309,19 +309,22 @@ def _run_fifo_vacuum(self, company=None):
...
         if company is None:
             company = self.env.company
+        done_products = self.browse()
         svls_to_vacuum = self.env['stock.valuation.layer'].sudo().search([
-            ('product_id', '=', self.id),
+            ('product_id', 'in', self.ids),
             ('remaining_qty', '<', 0),
             ('stock_move_id', '!=', False),
             ('company_id', '=', company.id),
         ], order='create_date, id')
```
Comment (sle-odoo) [inline addons/stock_account/models/product.py]: only get the oldest layer to vacuum per product

File: addons/stock_account/models/product.py [hunk 2]
```diff
@@ -309,19 +309,22 @@ def _run_fifo_vacuum(self, company=None):
...
         for svl_to_vacuum in svls_to_vacuum:
+            if svl_to_vacuum.product_id in done_products:
```
Comment (sle-odoo) [inline addons/stock_account/models/product.py]: do a single search to get all the fixing layers per layer to fix then sort them in python

**Comment by sle-odoo** [comment]
@nim-odoo we could have the same perf imp with this patch

``` diff
diff --git a/addons/stock_account/models/stock_move.py b/addons/stock_account/models/stock_move.py
index 86d871ff9c0..d934a4faa8d 100644
--- a/addons/stock_account/models/stock_move.py
+++ b/addons/stock_account/models/stock_move.py
@@ -263,8 +263,7 @@ class StockMove(models.Model):
                 stock_valuation_layers |= getattr(todo_valued_moves, '_create_%s_svl' % valued_type)()
                 continue

-
-        for svl in stock_valuation_layers:
+        for svl in stock_valuation_layers.with_context(active_test=False):
             if not svl.product_id.valuation == 'real_time':
                 continue
             if svl.currency_id.is_zero(svl.value):
@@ -276,7 +275,7 @@ class StockMove(models.Model):
         # For every in move, run the vacuum for the linked product.
         products_to_vacuum = valued_moves['in'].mapped('product_id')
         company = valued_moves['in'].mapped('company_id') and valued_moves['in'].mapped('company_id')[0] or self.env.company
-        for product_to_vacuum in products_to_vacuum:
+        for product_to_vacuum in products_to_vacuum.with_context(active_test=False):
             product_to_vacuum._run_fifo_vacuum(company)

         return res
```
but it's cleaner to remove/rename this field (note that it'll be removed in 13.3 once i finish the review of task-2178269)

**Comment by nim-odoo** [comment]
@sle-odoo I'd go for the less clean but stable patch, then!

**Comment by sle-odoo** [comment]
see https://github.com/odoo/odoo/pull/47478

**Pull request closed without merging**